File: src/ardrone_autonomy/src/hover_test.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Empty




def run():
	rospy.init_node('test_node')

	take_off = rospy.Publisher("/ardrone/takeoff", Empty, queue_size=1)
	land = rospy.Publisher("/ardrone/land", Empty, queue_size=1)

	while (take_off.get_num_connections() == 0):
		print("")

	take_off.publish()


	now = rospy.get_time()

	while (rospy.get_time() - now < 10):
		print(rospy.get_time() - now)

	while (land.get_num_connections() == 0):
		print("")

	print("1")
	land.publish()
	print("2")







if __name__ == '__main__':
	run()

File: src/ardrone_autonomy/src/hover_test.cpp
#include "ros/ros.h"
#include "std_msgs/Empty.h"
#include <ctime>

#include <sstream>



int main(int argc, char **argv){

	ros::init(argc,argv,"test_node");

	ros::NodeHandle n;

	std_msg::Empty msg;

	ros::Publisher takeOff = n.advertise<std_msgs::Empty>("/ardrone/takeoff",1);
	ros::
	while(ros::Publisher::getNumSubscribers() == 0) {}

	takeoff.publish(msg);

	
	
	return 0;

}